waku/common/tokenbucket.nim

```diff
@@ -0,0 +1,61 @@
+when (NimMajor, NimMinor) < (1, 4):
+  {.push raises: [Defect].}
+else:
+  {.push raises: [].}
+
+import chronos
+
+## This is an extract from chronos/ratelimit.nim due to the found bug in the original implementation.
+## Unfortunately that bug cannot be solved without harm the original features of TokenBucket class.
+## So, this current shortcut is used to enable move ahead with nwaku rate limiter implementation.
+type TokenBucket* = ref object
+  budget*: int
+  budgetCap: int
+  lastTimeFull: Moment
+  fillDuration: Duration
```

Comment:
May kindly add a comment wrt purpose of each attribute?

Reply:
Sure.